### PR TITLE
[CDAP-1032] Propagate cause of the exception with HandlerException

### PR DESCRIPTION
**Pull request by bdmogal**
Added cause to the `HandlerException`. Hard to debug without this.

### CODE REVIEW

**Comment by rp-julienguery** [comment]
LGTM

**Comment by bdmogal** [comment]
waiting for build to succeed.

File: src/main/java/co/cask/http/HttpResourceModel.java [hunk 1]
```diff
@@ -138,8 +138,7 @@ public HttpMethodInfo handle(HttpRequest request, HttpResponder responder, Map<S
           ("Problem accessing: %s. Reason: Method Not Allowed", request.getUri()));
       }
     } catch (Throwable e) {
-      throw new HandlerException(HttpResponseStatus.INTERNAL_SERVER_ERROR,
-                                 String.format("Error in executing path:"));
+      throw new HandlerException(HttpResponseStatus.INTERNAL_SERVER_ERROR, "Error in executing path", e);
```
Comment (chtyim) [inline src/main/java/co/cask/http/HttpResourceModel.java]: Includes the request URI and the request method as well.

Comment (chtyim) [inline src/main/java/co/cask/http/HttpResourceModel.java]: I mean in the error message.

Comment (bdmogal) [inline src/main/java/co/cask/http/HttpResourceModel.java]: Fixed.

**Comment by chtyim** [comment]
LGTM.

**Pull request merged**